x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/action/UpdateConnectorSyncJobErrorAction.java

```diff
@@ -0,0 +1,145 @@
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+import static org.elasticsearch.xcontent.ConstructingObjectParser.constructorArg;
+
+public class UpdateConnectorSyncJobErrorAction extends ActionType<AcknowledgedResponse> {
+
+    public static final UpdateConnectorSyncJobErrorAction INSTANCE = new UpdateConnectorSyncJobErrorAction();
+    public static final String NAME = "cluster:admin/xpack/connector/sync_job/update_error";
+    public static final String ERROR_EMPTY_MESSAGE = "[error] of the connector sync job cannot be null or empty";
+
+    private UpdateConnectorSyncJobErrorAction() {
+        super(NAME, AcknowledgedResponse::readFrom);
+    }
+
+    public static class Request extends ActionRequest implements ToXContentObject {
+        private static final ConstructingObjectParser<Request, String> PARSER = new ConstructingObjectParser<>(
+            "connector_sync_job_error_request",
+            false,
+            ((args, connectorSyncJobId) -> new Request(connectorSyncJobId, (String) args[0]))
+        );
+
+        static {
+            PARSER.declareString(constructorArg(), ConnectorSyncJob.ERROR_FIELD);
+        }
+
+        private final String connectorSyncJobId;
+
+        private final String error;
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.connectorSyncJobId = in.readString();
+            this.error = in.readString();
+        }
+
+        public Request(String connectorSyncJobId, String error) {
+            this.connectorSyncJobId = connectorSyncJobId;
+            this.error = error;
+        }
+
+        public static Request fromXContentBytes(String connectorSyncJobId, BytesReference source, XContentType xContentType) {
+            try (XContentParser parser = XContentHelper.createParser(XContentParserConfiguration.EMPTY, source, xContentType)) {
+                return UpdateConnectorSyncJobErrorAction.Request.fromXContent(parser, connectorSyncJobId);
+            } catch (IOException e) {
+                throw new ElasticsearchParseException("Failed to parse: " + source.utf8ToString());
+            }
+        }
+
+        public static UpdateConnectorSyncJobErrorAction.Request fromXContent(XContentParser parser, String connectorSyncJobId)
+            throws IOException {
+            return PARSER.parse(parser, connectorSyncJobId);
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            ActionRequestValidationException validationException = null;
+
+            if (Strings.isNullOrEmpty(connectorSyncJobId)) {
+                validationException = addValidationError(
+                    ConnectorSyncJobConstants.EMPTY_CONNECTOR_SYNC_JOB_ID_ERROR_MESSAGE,
+                    validationException
+                );
+            }
+
+            if (Strings.isNullOrEmpty(error)) {
```

Comment:
We can't reset the error to null?

Reply:
See comment above: I don't think we currently reset sync job errors.